pennylane/_qubit_device.py

```diff
@@ -283,6 +283,64 @@ def _collect_shotvector_results(self, circuit, counts_exist):
 
         return results
 
+    def _execute_new(self, circuit, **kwargs):
```

Comment:
Should we have `QubitDevice.execute` call on `QubitDevice._execute_new`?

Reply:
It was not necessary because we only use `batch_execute` and this should not be user facing.

Reply:
I added it because of the merge of PR 4.

Reply:
Got it!

It just came to mind though that `default.mixed` provides a custom `execute` method that uses `QubitDevice.execute` to determine info for the readout error. So basically `batch_execute` may use `_execute_new` that doesn't support the readout error.